mesonbuild/compilers/asm.py

```diff
@@ -0,0 +1,83 @@
+import typing as T
+
+from ..mesonlib import EnvironmentException, MachineChoice
+from .compilers import Compiler
+
+if T.TYPE_CHECKING:
+    from ..coredata import KeyedOptionDictType
+    from ..environment import Environment
+
+
+class NasmCompiler(Compiler):
+    language = 'asm'
+    id = 'nasm'
+
+    def needs_static_linker(self) -> bool:
+        return False
+
+    def get_always_args(self) -> T.List[str]:
+        if self.info.is_windows():
+            plat = 'win'
+        elif self.info.is_linux() or self.info.is_android():
```

Comment:
This is going to fail on the BSDs, Haiku, Solaris, and AIX, all of which we vaguely support, and all of which use elf.